blog/2017-12-11-end-to-end-testing-kubernetes-openshift.textile

```diff
@@ -0,0 +1,263 @@
+
+h3. Example 2
+
+Deploying a Wordpress and My SQL application to OpenShift from a Template file and testing it using Arquillian
+Kube Extension for OpenShift and  Fabric8 OpenShift Client.
+
+Source: "arquillian-testing-microservices/openshift-deployment-testing":https://github.com/arquillian-testing-microservices/openshift-deployment-testing
+
+h3. Example 3
+
+Building and deploying a sample SpringBoot GuestBook application with zero deployment configuration using
+"Fabric8 Maven Plugin":https://maven.fabric8.io/ and "Arquillian Kube Extension":https://github.com/arquillian/arquillian-kube .
+
+Fabric8 Maven Plugin aids in building Docker images and creating Kubernetes and OpenShift resource
+descriptors for the application that allows for a quick ramp-up with some opinionated defaults and Arquillian
+Kube Extension deploys the application from the generated resource descriptors and then executes deployment tests.
+
+Source: "arquillian-testing-microservices/zero-config-deployment-test":https://github.com/arquillian-testing-microservices/zero-config-deployment-test
+
+h2. Step By Step Guide To Writing Deployment Tests
+
+*For a more in-depth step by step coverage of deployment testing, continue reading below.*
+
+h3. Step 1: Setting Up Kubernetes/OpenShift Cluster
+
+One of the pre-requisites for Arquillian Kube Extension, is to have Kubernetes or OpenShift cluster running on
+your host machine.
+
+An easy way to setup Kubernetes or OpenShift cluster locally is using "Minikube":https://github.com/kubernetes/minikube and "Minishift":https://github.com/minishift/minishift respectively.
+
+h3. Step 2: Adding Arquillian Kube Dependencies to your project
+
+bq.. Arquillian Kube Extension provides a black box approach to testing your deployment that neither
+mutates the containers (by deploying, reconfiguring etc) nor the Kubernetes/OpenShift resources.
+
+It is used for immutable infrastructure and integration testing, wherein the test cases are meant to,
+consume and test the provided services and assert that the environment is in the expected state,
+providing you with the confidence that your application will work correctly when deployed on
+Kubernetes/OpenShift cluster.
+
+p. Before we can start writing our tests, we need to define a few dependencies as shown below:
+
+h4. Arquillian Cube BOM - Unified Dependencies
+
+bc(prettify).. <properties>
+    <version.arquillian_cube>${latest_released_version}</version.arquillian_cube>
+</properties>
+
+<dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>org.arquillian.cube</groupId>
+            <artifactId>arquillian-cube-bom</artifactId>
+            <version>${version.arquillian_cube}</version>
+            <scope>import</scope>
+            <type>pom</type>
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+
+h4. Arquillian Cube Requirement
```

Comment:
We mention this, but it's not explained why we need that library in the post.

Reply:
@lordofthejars, I am not sure of the answer, can you please help me with the answer.
>  why we need that library